### PR TITLE
Generalize UserAgreement component and use to introduce new genie warning/agreement

### PR DESCRIPTION
**Pull request by alisman**

### CODE REVIEW

File: src/appShell/App/usageAgreements/GenieAgreement.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import UsageAgreement from 'shared/components/UsageAgreement';
+import AppConfig from 'appConfig';
+import { getBrowserWindow } from 'cbioportal-frontend-commons';
+
+export const GENIE_PERSISTeNCE_KEY = 'genie-private-usage-agreement';
```
Comment (adamabeshouse) [inline src/appShell/App/usageAgreements/GenieAgreement.tsx:6]: `PERSISTeNCE`

File: src/appShell/App/usageAgreements/GenieAgreement.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+export function shouldShowGenieWarning() {
+    // we want to show a warning message on private cbioportal instances
+    // to prevent users from adding links in manuscripts
+    const showStudyViewWarning = ['cbioportal-genie-private'].includes(
```
Comment (adamabeshouse) [inline src/appShell/App/usageAgreements/GenieAgreement.tsx:11]: `showGenieWarning`

**Review by adamabeshouse — APPROVED** [review]
looks good after fixing typos

**Review by inodb — APPROVED** [review]
LGTM after fixing typos Adam mentioned

File: src/appShell/App/usageAgreements/GenieAgreement.tsx (hunk 2 above)
Comment (adamabeshouse) [reply src/appShell/App/usageAgreements/GenieAgreement.tsx:11]: not a big deal but its better for this to be named right

File: src/shared/components/UsageAgreement.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    handleModalHide() {
+        this.modalShow = false;
+    }
+    @autobind
+    handleChecked(item: string) {
+        this.checkedItems.set(item, !this.checkedItems.get(item));
+    }
+
+    render() {
+        console.log(this.props.useCheckboxes);
```
Comment (adamabeshouse) [inline src/shared/components/UsageAgreement.tsx:88]: log

File: src/shared/components/UsageAgreement.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    handleDismiss() {
+        this.show = false;
+        this.modalShow = false;
+
+        new expiredStorage().setItem(
+            this.props.persistenceKey,
+            'true',
+            this.expirationInSeconds
+        );
+        //localStorage.setItem(this.props.persistenceKey, 'true');
```
Comment (adamabeshouse) [inline src/shared/components/UsageAgreement.tsx:63]: comment

**Review by adamabeshouse — APPROVED** [review]
functionality looks good, just clean up the typo and its 💯

**Pull request merged**